plugins/out_stackdriver/stackdriver.c

```diff
@@ -341,6 +341,93 @@ static flb_sds_t get_str_value_from_msgpack_map(msgpack_object_map map,
+    msgpack_object *obj;
+    msgpack_unpacked result;
+    msgpack_unpacked_init(&result);
+    while (msgpack_unpack_next(&result, data, bytes, &off) == MSGPACK_UNPACK_SUCCESS) {
+        if (result.data.type != MSGPACK_OBJECT_ARRAY) {
+            continue;
+        }
+        if (result.data.via.array.size != 2) {
+            continue;
+        }
+        obj = &result.data.via.array.ptr[1];
+        if (obj->type != MSGPACK_OBJECT_MAP) {
+            continue;
+        }
+        msgpack_object_kv *kv = obj->via.map.ptr;
+        msgpack_object_kv *const kvend = obj->via.map.ptr + obj->via.map.size;
+        for (; kv < kvend; ++kv) {
+          if (kv->val.type == MSGPACK_OBJECT_MAP && kv->key.type == MSGPACK_OBJECT_STR
+          && strncmp (MONITORED_RESOURCE_KEY, kv->key.via.str.ptr, kv->key.via.str.size) == 0) {
+            msgpack_object subobj = kv->val;
+            msgpack_object_kv *p = subobj.via.map.ptr;
+            msgpack_object_kv *pend = subobj.via.map.ptr + subobj.via.map.size;
+            for (; p < pend; ++p) {
+              if (p->key.type != MSGPACK_OBJECT_STR || p->val.type != MSGPACK_OBJECT_MAP) {
+                continue;
+              }
+              if (strncmp("labels", p->key.via.str.ptr, p->key.via.str.size) == 0) {
+                  msgpack_object labels = p->val;
+                  msgpack_object_kv *q = labels.via.map.ptr;
+                  msgpack_object_kv *qend = labels.via.map.ptr + labels.via.map.size;
+                  int fields = 0;
+                  for (; q < qend; ++q) {
+                    if (q->key.type != MSGPACK_OBJECT_STR || q->val.type != MSGPACK_OBJECT_STR) {
+                        flb_plg_error(ctx->ins, "Key and value should be string in the %s/labels", MONITORED_RESOURCE_KEY);
+                    }
+                    ++fields;
+                  }
+                  if (fields > 0) {
+                    msgpack_pack_map(mp_pck, fields);
+                    q = labels.via.map.ptr;
+                    for (; q < qend; ++q) {
+                      if (q->key.type != MSGPACK_OBJECT_STR || q->val.type != MSGPACK_OBJECT_STR) {
+                          continue;
+                      }
+                      flb_plg_debug(ctx->ins, "[%s] found in the payload", MONITORED_RESOURCE_KEY);
+                      msgpack_pack_str(mp_pck, q->key.via.str.size);
+                      msgpack_pack_str_body(mp_pck, q->key.via.str.ptr, q->key.via.str.size);
+                      msgpack_pack_str(mp_pck, q->val.via.str.size);
+                      msgpack_pack_str_body(mp_pck, q->val.via.str.ptr, q->val.via.str.size);
+                    }
+                    msgpack_unpacked_destroy(&result);
+                    ret = 0;
+                    return ret;
+                  }
+              }
+            }
+          }
+        }
+    }
+
+    msgpack_unpacked_destroy(&result);
+    flb_plg_info(ctx->ins, "[%s] not found in the payload", MONITORED_RESOURCE_KEY);
```

Comment:
I think INFO log would make it too verbose. Do you think making it debug log would be reasonable?

Reply:
Done